app/src/main/java/com/rockthevote/grommet/data/db/model/RockyRequest.kt

```diff
@@ -0,0 +1,211 @@
+package com.rockthevote.grommet.data.db.model
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+
+@JsonClass(generateAdapter = false)
+data class RockyRequest(
+    @Json(name = "rocky_request")
+    val rockyRequest: RockyRequestBody
+)
+
+@JsonClass(generateAdapter = false)
+data class RockyRequestBody(
+    @Json(name = "lang")
+    val lang: String,
+    @Json(name = "phone_type")
+    val phoneType: String? = null, // TODO this is specified in attributes, but not in the example Json, IDK the source
+    @Json(name = "partner_id")
+    val partnerId: Int,
+    @Json(name = "opt_in_email")
+    val optInEmail: Boolean?,
+    @Json(name = "opt_in_sms")
+    val optInSms: Boolean?,
+    @Json(name = "opt_in_volunteer")
+    val optInVolunteer: Boolean?,
+    @Json(name = "partner_opt_in_sms")
+    val partnerOptInSms: Boolean,
+    @Json(name = "partner_opt_in_email")
+    val partnerOptInEmail: Boolean,
+    @Json(name = "partner_opt_in_volunteer")
+    val partnerOptInVolunteer: Boolean?,
+    @Json(name = "finish_with_state")
+    val finishWithState: Boolean?,
+    @Json(name = "created_via_api")
+    val createdViaApi: Boolean?,
+    @Json(name = "source_tracking_id")
+    val sourceTrackingId: String,
+    @Json(name = "partner_tracking_id")
+    val partnerTrackingId: String,
+    @Json(name = "geo_location")
+    val geoLocation: GeoLocation?,
+    @Json(name = "open_tracking_id")
+    val openTrackingId: String,
+    @Json(name = "voter_records_request")
+    val voterRecordsRequest: VoterRecordsRequest?
+)
+
+@JsonClass(generateAdapter = false)
+data class GeoLocation(
+    @Json(name = "lat")
+    val lat: Double,
+    @Json(name = "long")
+    val long: Double
+)
+
+@JsonClass(generateAdapter = false)
+data class VoterRecordsRequest(
+    @Json(name = "type")
+    val type: String?,
+    @Json(name = "generated_date")
+    val generatedDate: String,
+    @Json(name = "canvasser_name")
+    val canvasserName: String?,
+    @Json(name = "voter_registration")
+    val voterRegistration: VoterRegistration?
+)
+
+@JsonClass(generateAdapter = false)
+data class VoterRegistration(
+    @Json(name = "registration_helper")
+    val registrationHelper: RegistrationHelper?,
+    @Json(name = "date_of_birth")
+    val dateOfBirth: String,
+    @Json(name = "mailing_address")
+    val mailingAddress: Address?,
+    @Json(name = "previous_registration_address")
+    val previousRegistrationAddress: Address?,
+    @Json(name = "registration_address")
+    val registrationAddress: Address,
+    @Json(name = "registration_address_is_mailing_address")
+    val registrationAddressIsMailingAddress: Boolean,
+    @Json(name = "name")
+    val name: Name,
+    @Json(name = "previous_name")
+    val previousName: Name?,
+    @Json(name = "gender")
+    val gender: String, // TODO this is non-null in the old API request, but not required per API docs, which is correct?
```

Comment:
This is inferred based on their prefix (mr/ms/etc), which is required, so I would say this is required
.

Reply:
Sounds good, I'll make it mandatory